client/src/main/java/com/alibaba/nacos/client/naming/utils/Chooser.java

```diff
@@ -138,7 +138,7 @@ public void refresh() {
             }
 
             double doublePrecisionDelta = 0.0001;
-            if (index != 0 && !(Math.abs(weights[index - 1] - 1) < doublePrecisionDelta)) {
+            if (index == 0 || (Math.abs(weights[index - 1] - 1) < doublePrecisionDelta)) {
```

Comment:
Seems it should be ```if (!(index == 0 || (Math.abs(weights[index - 1] - 1) < doublePrecisionDelta)))```

Reply:
yes its my fault i forget return in this part
just according to p3c (! is not clear for code)